### ISSUE TITLE
" invalid suffix 'x' on integer constant" -- issue building with ubuntu releases from last 3 years

### ISSUE THREAD

**Issue by rrnewton** [project-dune/dune#2]
I tried a quick build of dune on Ubuntu 14.04, 16.04, and 17.04 machines (gcc 4.9.4, 5.4, 6.3,  kernel 3.19, 4.4, 4.10), and in all cases `dune_req` reported "yes", but I had the same error compile:

```
  CC [M]  /nfs/home/rrnewton/temp/dune/kern/vmx.o
<command-line>:0:9: error: invalid suffix "x" on integer constant
/nfs/home/rrnewton/temp/dune/kern/compat.h:78:53: note: in expansion of macro ‘DO_FORK’
 static do_fork_hack __dune_do_fork = (do_fork_hack) DO_FORK;
                                                     ^
```

**Comment by rrnewton** [comment]
Oh, that may have been the consequence of earlier, non-fatal failures:
```
cat: /boot/System.map-4.10.0-20-generic: Permission denied
cat: /boot/System.map-4.10.0-20-generic: Permission denied
cat: /boot/System.map-4.10.0-20-generic: Permission denied
cat: /boot/System.map-4.10.0-20-generic: Permission denied
```
As a quick hack, building as sudo gets it to this error instead:
```  CC [M]  /home/rrnewton/temp/dune/kern/vmx.o
/home/rrnewton/temp/dune/kern/vmx.c: In function ‘ept_sync_individual_addr’:
/home/rrnewton/temp/dune/kern/vmx.c:184:12: error: ‘VMX_EPT_EXTENT_INDIVIDUAL_ADDR’ undeclared (first use in this function)
   __invept(VMX_EPT_EXTENT_INDIVIDUAL_ADDR,
```

**Comment by kkaffes** [comment]
Did you encounter any errors when using Ubuntu 16.04?

**Comment by rrnewton** [comment]
WOW, you're right -- the data point I left out is the one that works.  `sudo make` works for me on Ubuntu 16.04, Linux 4.4.0.

Should it be able to build without sudo?  The README doesn't say so.

**Comment by kkaffes** [comment]
No, sudo is required for building. I haven't been able to reproduce the issue in Ubuntu 14.04. Did you try to 
run `sudo make clean` before compiling again?

Regarding Ubuntu 17.04, the `VMX_EPT_EXTENT_INDIVIDUAL_ADDR` declaration was removed from kernel version  4.10 and according to the KVM commit history

> Bit 6 in EPT vmexit's exit qualification is not defined in SDM, so remove it.

so it might make sense to remove it altogether.

**Comment by rrnewton** [comment]
Ah, I tried again with fresh checkouts on two different 14.04 machines.  But they're both part of an HPC cluster so they may have been made nonstandard in certain ways by the sysadmin.

Let's see... it should build in a VM (or even docker container) right?  Even if Dune can't actually be used from within a VM.  I could use that to try to reproduce on a fresh 14.04 and send the vagrantfile...

**Comment by kkaffes** [comment]
I just built a fresh checkout in an Ubuntu 14.04 VM without issues. If you are able to reproduce it in a VM, please send me the vagrantfile and I will look into it.

**Comment by prekageo** [comment]
`sudo` is not required to `make`. Check the README of https://github.com/ix-project/ix. Specifically, you just have to do
```
sudo chmod +r /boot/System.map-`uname -r`
```

**Comment by kkaffes** [comment]
Closing the issue due to inactivity.

**Comment by Shihao-Zhong** [comment]
Hi @kkaffes, I am facing the same issue in Cloudlab Ubuntu16.04(https://www.cloudlab.us/show-profile.php?uuid=3dd079eb-97d0-11e6-ac8c-90e2ba22fee4) with kernal 4.4.0 to build the dune from branch linux5 and master. Could you please share me some insight about how you guys fixed that in 16.04?

Linux5: 
/users/szhong01/dune/kern/vmx.c: In function ‘construct_eptp’:
/users/szhong01/dune/kern/vmx.c:766:9: error: ‘VMX_EPTP_MT_WB’ undeclared (first use in this function)
  eptp = VMX_EPTP_MT_WB | VMX_EPTP_PWL_4;
         ^
/users/szhong01/dune/kern/vmx.c:766:26: error: ‘VMX_EPTP_PWL_4’ undeclared (first use in this function)
  eptp = VMX_EPTP_MT_WB | VMX_EPTP_PWL_4;
                          ^
/users/szhong01/dune/kern/vmx.c: In function ‘dune_sys_clone’:
/users/szhong01/dune/kern/vmx.c:1238:9: error: implicit declaration of function ‘dune_do_fork’ [-Werror=implicit-function-declaration]
  return dune_do_fork(clone_flags, newsp, &regs, 0, parent_tid, child_tid,
         ^
/users/szhong01/dune/kern/vmx.c: In function ‘__vmx_enable’:
/users/szhong01/dune/kern/vmx.c:1668:6: error: implicit declaration of function ‘dune_tboot_enabled’ [-Werror=implicit-function-declaration]
  if (dune_tboot_enabled())


Master: 
/apic.c: In function ‘dune_apic_send_ipi_x2’:
/apic.c:103:2: error: implicit declaration of function ‘x2apic_wrmsr_fence’ [-Werror=implicit-function-declaration]
  x2apic_wrmsr_fence();
